src/plugins/customer-order/generate-order/generate-order.resolver.ts

```diff
@@ -0,0 +1,112 @@
+import { GenerateCustomerOrderService } from './generate-customer-order.service';
+import { Args, Resolver, Mutation } from '@nestjs/graphql';
+import {
+    OrderService, Transaction,
+    Ctx, RequestContext, UnauthorizedError,
+    UserInputError, Order,
+    ID, ProductVariant, InternalServerError, IllegalOperationError, ErrorResultUnion
+} from '@vendure/core';
+import { ActiveOrderResult } from '@vendure/common/lib/generated-shop-types'
+
+@Resolver('GenerateOrder')
+export class GenerateOrderResolver {
+
+    constructor(
+        private orderService: OrderService,
+        private generateCustomerOrderService: GenerateCustomerOrderService
+    ) { }
+
+    @Transaction()
+    @Mutation()
+    async generateOrderBySkus(
+        @Ctx() ctx: RequestContext,
+        @Args() args: any,
+    ): Promise<ActiveOrderResult | any> {
+
+        if (!ctx.session) {
+            throw new UnauthorizedError();
+        }
+
+        let jsonOrder = JSON.parse(args.skus);
+        let skus: Array<String> = []
+        jsonOrder.map((sku: any) => {
+            skus.push(sku.code);
+        });
+        if (skus.length == 0) {
+            throw new UserInputError('Solicitud incorrecta!');
+        }
+        let search: Array<ProductVariant>;
+        try {
+            search = await this.generateCustomerOrderService.findProductVariantsBySKU(ctx, skus);
+        } catch (e) {
+            search = [];
+        }
+
+        let order: Order | any | undefined;
+
+        if (search.length > 0 && this.generateCustomerOrderService.isStockAvailable(search)) {
+            try{
+                order = await this.orderService.getActiveOrderForUser(ctx, ctx.session.user?.id as ID);
+                if (!order) {
+                    order = await this.orderService.create(ctx,ctx.session.user?.id as ID);
+                }
+            }catch(e){console.log("Error to get or create order");}
```

Comment:
Seria bueno logguear el error...  o incluso es un console.error
```suggestion
            }catch(e){console.error("Error to get or create order", e);}
```